src/App.tsx

```diff
@@ -16,80 +17,136 @@ export const goodsFromServer = [
 ];
 
 enum SortType {
-  NONE,
-  ALPABET,
-  LENGTH,
+  NONE = 'none',
+  ALPABET = 'alphabet',
+  LENGTH = 'length',
 }
 
-type ReorderOptions = {
-  sortType: SortType,
+type State = {
   isReversed: boolean,
+  sortBy: string,
+  goods: string[],
 };
 
-// Use this function in the render to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortBy: '',
+    goods: goodsFromServer,
+  };
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  sortAlpha = () => {
+    this.setState({
+      sortBy: SortType.ALPABET,
+    });
+  };
 
-  return visibleGoods;
-}
+  sortLength = () => {
+    this.setState({
+      sortBy: SortType.LENGTH,
+    });
+  };
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      goods: goodsFromServer,
+      isReversed: false,
+      sortBy: '',
+    });
+  };
+
+  render() {
+    const {
+      isReversed,
+      sortBy,
+      goods,
+    } = this.state;
+
+    const visibleGoods = [...goods];
+
+    visibleGoods.sort((a, b) => {
+      switch (sortBy) {
+        case 'alphabet':
+          return a.localeCompare(b);
+
+        case 'length':
+          return a[sortBy] - b[sortBy];
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
```

Comment:
винеси це в тіло компоненту
```suggestion
    const visibleGoods = [...goodsFromServer].sort((a, b) => {
      switch (sortBy) {
        case SortType.ALPABET:
          return a.localeCompare(b);

        case SortType.LENGTH:
          return a[sortBy] - b[sortBy];

        default:
          return 0;
      }
    });

    if (isReversed) {
      visibleGoods.reverse();
    }
```

Reply:
Не зовсім зрозумів, як саме це винести в тіло компонента.